edgepi-relay.js

```diff
@@ -1,44 +1,54 @@
 module.exports = function (RED) {
-    const rpc = require('@edgepi-cloud/edgepi-rpc')
-  
-    function RelayNode(config) {
-      // Create new node instance with user config
-      RED.nodes.createNode(this, config);
-      const node = this;
-      const ipc_transport = "ipc:///tmp/edgepi.pipe"
-      const tcp_transport = `tcp://${config.tcpAddress}:${config.tcpPort}`
-      const transport = (config.transport === "Network") ? tcp_transport : ipc_transport;
-      const relayState = (config.relayState)
-  
-      // init new relay instance
-      const relay = new rpc.RelayService(transport)
-  
-      if (relay){
-        console.info("Relay node initialized on:", transport);
-        node.status({fill:"green", shape:"ring", text:"relay initialized"});
-      }
-  
-      // Input event listener
-      node.on('input', async function(msg,send,done){
-        node.status({fill:"green", shape:"dot", text:"input recieved"});
-        try{
-          const response = await relay[relayState]();
-          msg.payload = response;
-        }
-        catch(error){
+  const rpc = require("@edgepi-cloud/edgepi-rpc");
+
+  function RelayNode(config) {
+    RED.nodes.createNode(this, config);
+    const node = this;
+    let relayState = config.relayState;
+
+    initializeNode(config).then((relay) => {
+      node.on("input", async function (msg, send, done) {
+        node.status({ fill: "green", shape: "dot", text: "input recieved" });
+        try {
+          relayState = msg.payload ?? relayState;
+          const stateStr = relayState === true ? "closeRelay" : "openRelay";
+          msg = {payload: await relay[stateStr]()};
```

Comment:
This is just for my understanding but can you explain `relay[stateStr]()`? Is this part updating the configuration then returning the associated boolean when it's successfully updated using `stateStr`?

Reply:
`relay[stateStr]()` is a function call. E.g. if `stateStr` is `openRelay`, this would call `openRelay()`.